ichnaea/scripts/apikey.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+"""
+Create and manipulate API keys.
+"""
+
+import argparse
+import uuid
+import sys
+
+from sqlalchemy import select
+from sqlalchemy.dialects.mysql import insert
+from sqlalchemy.exc import IntegrityError
+
+from ichnaea.api.key import API_KEY_COLUMN_NAMES, Key
+from ichnaea.models.api import ApiKey
+from ichnaea.db import configure_db, db_worker_session
+from ichnaea.util import print_table
+
+
+def create_api_key(key):
+    """Create a new api key."""
+    key = key or str(uuid.uuid4())
+
+    db = configure_db("rw")
+    with db_worker_session(db) as session:
+        try:
+            session.execute(
+                insert(ApiKey.__table__).values(
+                    valid_key=key,
+                    allow_fallback=False,
+                    allow_locate=True,
+                    allow_region=True,
```

Comment:
This hard-codes most of the values except the key. I wasn't sure whether to expose everything or not. I didn't want to create an annoying interface for it and wasn't sure what the predominant use cases would be. 

Maybe it's better to have subcommands for specific scenarios? Maybe "build an api key from this ini file" or something like that?

I figure we can defer that to when it comes up.

Reply:
I'm surprised the models don't have code-side or database-side defaults defined, which is really common in the Django world, but not in the SQLAlchemy world apparently. I'd prefer that the default values live on ``ApiKey`` eventually. It doesn't need to be in this PR.

I think we should use ops defaults for API keys (see private repo https://github.com/mozilla-services/cloudops-docs/tree/master/Services/Location#adding-new-api-key). That would add:

* ``maxreq=100000``
* ``store_sample_locate=100``
* ``store_sample_submit=100``

I think this would fulfill current needs, until we implement commercial user features like #890.

Reply:
I don't understand sqlalchemy or how things are implemented. For example, there's an ApiKey and also a Key. One is db-backed and the other is cache-backed. Also, there is a factory for each model in the tests. So if you adjust one, you have to adjust it in four different places. That's curious. Why have two models for the same data? How do you know which one to use in the code? Why is there no abstraction layer for manipulating api key data? Why are default values strewn about? Anyhow... I figured I'd look into that at some later point.

Using ops defaults  is interesting, but I'm hesitant. This api key cli is currently just for the local dev environment. If we add ops defaults, I worry that suggests it should be used as a cli in production. I don't want to mix those two use cases because I suspect they have very different needs. That can get figured out in #846 .

Having said that, we should add the two store things because we need that for development. I'll add that now.